### PR TITLE
[Form] add option to render NumberType as type="number"

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #8106
| License       | MIT
| Doc PR        | symfony/symfony-docs#10997

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/NumberType.php [hunk 1]
```diff
@@ -26,10 +28,21 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $builder->addViewTransformer(new NumberToLocalizedStringTransformer(
             $options['scale'],
             $options['grouping'],
-            $options['rounding_mode']
+            $options['rounding_mode'],
+            $options['html5'] ? 'en' : null
```
Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Core/Type/NumberType.php]: shouldnt we expose the locale option instead? Makes sense for number formatting being locale aware.

Then we can be future proof, and render type=number for any supporting locale (`en === locale` of check from Intl maybe)

Comment (xabbuh) [reply src/Symfony/Component/Form/Extension/Core/Type/NumberType.php]: I am not sure. I mean I would not expose it here to solve this particular use case. Being able to configure the locale to be used should then IMO be made possible for all form types that use locale depending formatters. But this then only makes sense if you need to change the locale per type. Not sure that there are any use cases for that. To set it globally using the `Locale` class IMO sounds better then introducing yet another config option.

Comment (ro0NL) [reply src/Symfony/Component/Form/Extension/Core/Type/NumberType.php]: > Not sure that there are any use cases for that.

indeed, i guess not :D in practice you'll always input normalized numbers (in a single locale).

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Bridge/Twig/Tests/Extension/AbstractBootstrap3LayoutTest.php [hunk 2]
```diff
@@ -2089,6 +2089,22 @@ public function testNumber()
         );
     }
 
+    public function testRenderNumberWithHtml5NumberType()
+    {
+        $form = $this->factory->createNamed('name', 'Symfony\Component\Form\Extension\Core\Type\NumberType', 1234.56, [
```
Comment (javiereguiluz) [inline src/Symfony/Bridge/Twig/Tests/Extension/AbstractBootstrap3LayoutTest.php]: In addition to `1234.56`, would it make sense to test what happens with `1,234.56` ? Thanks.

Comment (xabbuh) [reply src/Symfony/Bridge/Twig/Tests/Extension/AbstractBootstrap3LayoutTest.php]: As far as I know that's not something the client would send in this scenario. Or do you mean having a test that makes sure we get a transformation failure?

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/NumberToLocalizedStringTransformer.php [hunk 3]
```diff
@@ -214,7 +216,7 @@ public function reverseTransform($value)
      */
     protected function getNumberFormatter()
     {
-        $formatter = new \NumberFormatter(\Locale::getDefault(), \NumberFormatter::DECIMAL);
+        $formatter = new \NumberFormatter(null === $this->locale ? \Locale::getDefault() : $this->locale, \NumberFormatter::DECIMAL);
```
Comment (vudaltsov) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/NumberToLocalizedStringTransformer.php]:
```suggestion
        $formatter = new \NumberFormatter($this->locale ?? \Locale::getDefault(), \NumberFormatter::DECIMAL);
```

**Review by vudaltsov — APPROVED** [review]

File: src/Symfony/Bridge/Twig/Tests/Extension/AbstractBootstrap3LayoutTest.php (hunk 2 above)
Comment (xabbuh) [reply src/Symfony/Bridge/Twig/Tests/Extension/AbstractBootstrap3LayoutTest.php]: @javiereguiluz I added such a test case to make sure that we provide a nice error message when you try to use the `grouping` option for HTML5 widgets. Good for you?

Comment (javiereguiluz) [reply src/Symfony/Bridge/Twig/Tests/Extension/AbstractBootstrap3LayoutTest.php]: Yes! Thank you.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**